src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +42,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
+        thisx->draw = BgMoriBigst_Draw;
+        if (Flags_GetClear(globalCtx, thisx->room) && (PLAYER->actor.posRot.pos.y > 700.0f)) {
+            if (Flags_GetSwitch(globalCtx, (thisx->params >> 8) & 0x3F)) {
+                BgMoriBigst_SetupDone(this, globalCtx);
+            } else {
+                BgMoriBigst_SetupStalfosFight(this, globalCtx);
+            }
+        } else {
+            BgMoriBigst_SetupNoop(this, globalCtx);
+        }
+    }
+}
+
+void BgMoriBigst_SetupNoop(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetupAction(this, NULL);
+}
+
+void BgMoriBigst_SetupStalfosFight(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* stalfos;
+
+    BgMoriBigst_SetupAction(this, BgMoriBigst_StalfosFight);
+    Flags_UnsetClear(globalCtx, this->dyna.actor.room);
+    stalfos = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_EN_TEST, 209.0f, 827.0f,
+                                 -3320.0f, 0, 0, 0, 1);
+    if (stalfos != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Second Stalfos failure
+        osSyncPrintf("Warning : 第２スタルフォス発生失敗\n");
+    }
+    Flags_SetClear(globalCtx, this->dyna.actor.room);
+}
+
+void BgMoriBigst_StalfosFight(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->dyna.actor.initPosRot.rot.z == 0) &&
+        ((this->dyna.actor.initPosRot.pos.y - 5.0f) <= PLAYER->actor.posRot.pos.y)) {
+        BgMoriBigst_SetupFall(this, globalCtx);
+        func_800800F8(globalCtx, 0xC94, 0x48, &this->dyna.actor, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/BgMoriBigst_Init.s")
+void BgMoriBigst_SetupFall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetupAction(this, BgMoriBigst_Fall);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/BgMoriBigst_Destroy.s")
+void BgMoriBigst_Fall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->dyna.actor);
+    if (this->dyna.actor.posRot.pos.y <= this->dyna.actor.initPosRot.pos.y) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        BgMoriBigst_SetupLanding(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        func_800800F8(globalCtx, 0x3FC, 8, &this->dyna.actor, 0);
+        func_8002DF38(globalCtx, NULL, 0x3C);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/func_808A0DDC.s")
+void BgMoriBigst_SetupLanding(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Camera** cameras = globalCtx->cameraPtrs;
+    s32 quake;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Bigst/func_808A0E04.s")
+    BgMoriBigst_SetupAction(this, BgMoriBigst_Landing);
+    this->waitTimer = 18;
+    quake = Quake_Add(cameras[globalCtx->activeCamera], 3);
```

Comment:
Is the `cameras` required here? If so that's really unfortunate
```suggestion
    quake = Quake_Add(ACTIVE_CAM, 3);
```